### PR TITLE
Nn 479 auto allocation provisional

### PR DESCRIPTION
**Pull request by steverendell**
Includes some fixes to get it working hosted in KW service

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/hmpps/keyworker/model/OffenderKeyworker.java [hunk 1]
```diff
@@ -24,7 +24,10 @@
 @EqualsAndHashCode(of = {"offenderNo", "staffId", "assignedDateTime"})
 @EntityListeners(AuditingEntityListener.class)
 public class OffenderKeyworker {
-
+    /*
+public  String testMethod() {
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/digital/hmpps/keyworker/model/OffenderKeyworker.java]: Is this needed?

File: src/main/java/uk/gov/justice/digital/hmpps/keyworker/repository/OffenderKeyworkerRepository.java [hunk 2]
```diff
@@ -20,8 +24,18 @@
 
     List<OffenderKeyworker> findByActiveAndOffenderNoIn(boolean isActive, Collection<String> offenderNo);
     List<OffenderKeyworker> findByActiveAndOffenderNo(boolean isActive, String offenderNo);
+    //List<OffenderKeyworker> findByAllocationTypeAndOffenderNoAndStaffId(AllocationType type, String offenderNo, Long staffId);
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/digital/hmpps/keyworker/repository/OffenderKeyworkerRepository.java]: Commented out?

File: src/main/java/uk/gov/justice/digital/hmpps/keyworker/model/OffenderKeyworker.java (hunk 1 above)
Comment (steverendell) [reply src/main/java/uk/gov/justice/digital/hmpps/keyworker/model/OffenderKeyworker.java]: oops!

File: src/main/resources/db/migration/V1_1__offender_key_worker.sql [hunk 3]
```diff
@@ -27,7 +27,7 @@ CREATE TABLE OFFENDER_KEY_WORKER
 
   CONSTRAINT OFFENDER_KEYWORKER_UNIQUE UNIQUE (OFFENDER_NO, STAFF_ID, ASSIGNED_DATE_TIME),
 
-  CONSTRAINT OFFENDER_KEY_WORKER_C1 CHECK (ALLOC_TYPE IN ('A','M'))
+  CONSTRAINT OFFENDER_KEY_WORKER_C1 CHECK (ALLOC_TYPE IN ('A','M','P'))
```
Comment (Mjwillis) [inline src/main/resources/db/migration/V1_1__offender_key_worker.sql]: @steverendell forgot to see this, you can't change flyway files - they are already in the db, and they will fail.  Hence why KW service is down.

**Pull request merged**